baseline-error-prone/src/test/java/com/palantir/baseline/errorprone/FallThroughTest.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.baseline.errorprone;
+
+import com.google.common.collect.ImmutableList;
+import com.google.errorprone.CompilationTestHelper;
+import com.google.errorprone.bugpatterns.FallThrough;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.condition.DisabledForJreRange;
+import org.junit.jupiter.api.condition.JRE;
+
+public class FallThroughTest {
+
+    @Test
+    @DisabledForJreRange(max = JRE.JAVA_13)
+    public void testSwitchExpression() {
+        CompilationTestHelper compilationHelper = CompilationTestHelper.newInstance(FallThrough.class, getClass())
+                .setArgs(ImmutableList.of("--enable-preview", "--release", "14"));
+
+        compilationHelper
+                .addSourceLines(
+                        "Test.java",
+                        "class Test {",
+                        "  static void foo(int value) {",
+                        "    switch (value) {",
+                        "      case 42 -> {}",
+                        "      // BUG: Diagnostic matches: X",
+                        "      default -> {}",
+                        "    };",
+                        "  }",
+                        "}")
+                .expectErrorMessage("X", input -> input.contains("Execution may fall through from the previous case"))
```

Comment:
I had to use `Diagnostic matches` and `expectErrorMessage` because otherwise the diagnostic message matches the magic comment that will suppress the check.
https://github.com/google/error-prone/blob/v2.4.0/core/src/main/java/com/google/errorprone/bugpatterns/FallThrough.java#L45-L46